#166272297 Login page

EMacco opened the pull request:
#### What does this PR do?
Allows user to be able to login to author's haven

#### Description of Task to be completed?
- Validate user email and password 
- Display global error messages 
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login 

#### How should this be manually tested?
Navigate to / and click on the `Sign In` button on the top right and fill in the required details

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[166272297](166272297)

#### Screenshots (if appropriate)
<img width="1440" alt="Screenshot 2019-07-12 at 4 43 14 PM" src="https://user-images.githubusercontent.com/20377428/61146818-cfecc980-a4d2-11e9-93b5-af0248d6f2f9.png">
<img width="1440" alt="Screenshot 2019-07-12 at 4 43 36 PM" src="https://user-images.githubusercontent.com/20377428/61146830-d713d780-a4d2-11e9-8225-192c1c13b113.png">
<img width="1440" alt="Screenshot 2019-07-12 at 4 42 08 PM" src="https://user-images.githubusercontent.com/20377428/61146848-e2670300-a4d2-11e9-9908-32d1f422a733.png">


#### Questions:
N/A

despeauxz committed:
ch-[166272289]: setup redux

- setup the provider from react-redux
- configured the store
- initialized an empty reducer

[#166272289]

despeauxz committed:
ch-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
ch-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
ch-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
ch-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
chore-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
Merge branch 'develop' into 166272289-ch-setup-redux

despeauxz committed:
chore-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

Finishes #167164772]

despeauxz committed:
chore-[166272289]: setup redux

- setup the provider from react-redux
- configure the store
- initialize an emoty reducer
- setup PR template

[#166272289]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

 [Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

 [Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

 [Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

 [Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

 [Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader
[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

despeauxz committed:
feat-[167164772]: Common Components

Create the following components
- Create the following common components
- Header
- NavBar
- Footer
- Input
- Button
- Modal
- Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
feat-[167164772]: Common Components

Create the following components:
 - Create the following common components:
 - Header
 - NavBar
 - Footer
 - Input
 - Button
 - Modal
 - Preloader

[Finishes #167164772]

EMacco committed:
ft-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
ft-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

GerrardE reviewed (dismissed):
Good Job

fidelisojeah commented on src/config/axios.js:
```diff
@@ -0,0 +1,33 @@
+      (status === 400 &&
+        data.errors.global === 'Invalid token supplied: format Bearer <token>')
+    ) {
+      window.location.reload();
```
wait what?

fidelisojeah commented on src/helpers/decodeToken.js:
```diff
@@ -0,0 +1,11 @@
+import jwt from 'jsonwebtoken';
```
please get rid of all this.

The worst you could do is:

stringify the values from the backend and store the values in the localstorage.

but don't decode the token in the frontend

fidelisojeah replied:
you could also send back the `expires in` part from the backend if you want (raise a ticket for that)

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]

fidelisojeah commented on __test__/components/commons/Header.test.js:
```diff
@@ -1,10 +1,13 @@
 import React from 'react';
```
What exactly are you testing in this file?

fidelisojeah commented on __test__/login.test.js:
```diff
@@ -1,23 +1,35 @@
+import { BrowserRouter, Switch } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import auth from '@reducers/auth';
+import toJson from 'enzyme-to-json';
```
Please get rid of this.

There's a way to render the wrapper to json globally

EMacco committed:
feat-[166272297]: Login Page

- Validate user email and password
- Display global error messages
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login

[Finishes #166272297]
